### ISSUE TITLE
Handle skipCertVerification field for pipelines fetch

### ISSUE THREAD

**Issue by brianxjx** [kabanero-io/kabanero-operator#633]
Setting 
```skipCertVerification: true```
for pipelines.https seems to have no effect.

Steps to reproduce:
install default Kab instance
create and deploy Nginx container hosting pipelines tarball
edit Kab CR to point to Nginx container route, including skipCertVerification: true
check stack controller logs to see `x509: certificate signed by unknown authority`
listing pipelines returns empty as original defaults have been removed (correctly) but newly config'd pipelines have not loaded.

**Comment by kaczyns** [comment]
Looking at the code....
https://github.com/kabanero-io/kabanero-operator/blob/310160e8d1b9df3410c34d54d017baf015ecbb32/pkg/controller/stack/archive.go#L56
seems to always pass `false` to the cache which always causes certificates to be verified.

To get this to work, we'd have to pass in more than just the URL to this method.  I think back in stack_controller.go, we'd have to add more than just the `url` to the `pipelineUseMapValue`:
https://github.com/kabanero-io/kabanero-operator/blob/310160e8d1b9df3410c34d54d017baf015ecbb32/pkg/controller/stack/stack_controller.go#L385

the `pipelineUseMapValue` is an extended form of the `PipelineStatus`.  So perhaps we can just add the whole `HttpsProtocolFile` to the `PipelineStatus` instead of just the URL part.  It would change the structure of the Stack CR status somewhat.  Or could just add it directly to the `pipelineUseMapValue` so that it can be read out later and passed to `GetManifests()`.

**Comment by brianxjx** [comment]
Fyre 4.3 cluster w/ Kabanero 0.8.0-rc.2
`oc get kabanero kabanero -o yaml`
```
  stacks:
    pipelines:
    - gitRelease: {}
      https:
        skipCertVerification: true
        url: https://pipelines-index-latest-kabanero.apps.wharfrat.os.fyre.ibm.com/default-kabanero-pipelines.tar.gz
      id: default
      sha256: d4d825415e933a2ceabd9e5be94f8bf2441abc85b666d83b5ff94b2b5428dc59
```

`oc get stack java-microprofile -o yaml`
```
spec:
  name: java-microprofile
  versions:
  - images:
    - id: Eclipse MicroProfile®
      image: docker.io/kabanerobeta/java-microprofile
    pipelines:
    - gitRelease: {}
      https:
        skipCertVerification: true
        url: https://pipelines-index-latest-kabanero.apps.wharfrat.os.fyre.ibm.com/default-kabanero-pipelines.tar.gz
      id: default
      sha256: d4d825415e933a2ceabd9e5be94f8bf2441abc85b666d83b5ff94b2b5428dc59
```

**Comment by kaczyns** [comment]
While I was researching the fix for this, I stumbled on something that seems kind of minor but can really mess up how the stack controller manages the tekton artifacts.

There is this thing called the `pipelineUseMapKey` that is basically the key to a map that manages whether a stack version uses a particular pipeline version (meaning a pipeline zip with a particular URL and digest). That key can either contain a URL, or a struct of GitHub release information.  The value of this map then becomes the content of the pipeline zip and a bunch of other meta-data about how it is used by the various stack versions.

When we introduced GHE support, the `skipCertVerification` boolean ended up in the `pipelineUseMapKey`.  If you had two stack versions, which only differed in `skipCertVerification`, this would cause the stack controller to try and load and apply that pipeline twice.  However when one version got removed from the Stack CR instance, the results would be non-deterministic, because the stack controller would think it had to delete those artifacts for the version that just went away, but then keep them for the other... but they have two different map keys.

When I fix this, I'm going to remove the cert verification from the key if I can.  The cert verification information can be kept on the side and used if we need to retrieve the pipeline zip from either the web, or from a GitHub/GHE release.

The cert verification information should probably also be removed from the Stack CR status, but that's less important.
